scripts/mkcloudhost/routed.cloud

```diff
@@ -12,10 +12,20 @@ fi
 : ${reposerver:=clouddata.cloud.suse.de}
 : ${nfsserver:=clouddata.cloud.suse.de}
 export reposerver nfsserver
-export net_admin=$(cloudadminnet $n)
-export net_ironic=$(cloudironicnet $n)
-export net_public=$(routedcloudpublicnet $n)
-export adminnetmask=255.255.255.0
+if (( $want_ipv6 > 0 )); then
+    prefix=$(ipv6prefix $n)
+    export net_admin=$prefix
+    export net_public=$prefix
+    export net_ironic=$prefix
```

Comment:
I haven't testing using just 1 network for admin and public.. It should work, but we might need to confirm that network ranges don't overlap or we'll get some validation issues. If my laptop _ever_ finishes cacheing (rsync) the repo for local use I'll give it a whirl... just to confirm.

A quick look at the network.json from automation/scripts/lib/ipv6/network.json the admin and public dhcp ranges might overlap:

                  

>               "admin": {
>                     "add_bridge": false,
>                     "broadcast": "",
>                     "conduit": "intf0",
>                     "mtu": 1500,
>                     "netmask": "64",
>                     "ranges": {
>                         "admin": {
>                             "end": "fd00:0:0:3:5054:ff:fe77:7771",
>                             "start": "fd00:0:0:3:5054:ff:fe77:7770"
>                         },
>                         "dhcp": {
>                             "end": "fd00:0:0:3::80",
>                             "start": "fd00:0:0:3::21"
>                         },
>                         "host": {
>                             "end": "fd00:0:0:3::160",
>                             "start": "fd00:0:0:3::81"
>                         },
>                         "switch": {
>                             "end": "fd00:0:0:3::250",
>                             "start": "fd00:0:0:3::241"
>                         }
>                     },
>                     "router": "fd00:0:0:3::1",
>                     "router_pref": 10,
>                     "subnet": "fd00:0:0:3::0",
>                     "use_vlan": false,
>                     "vlan": 100
>                 },

And

>                 "public": {
>                     "add_bridge": false,
>                     "broadcast": "",
>                     "conduit": "intf1",
>                     "netmask": "64",
>                     "ranges": {
>                         "host": {
>                             "end": "fd00:0:0:1::127",
>                             "start": "fd00:0:0:1::2"
>                         }
>                     },
>                     "router": "fd00:0:0:1::1",
>                     "router_pref": 5,
>                     "subnet": "fd00:0:0:1::0",
>                     "use_vlan": true,
>                     "vlan": 300
>                 },
> 

Once the prefixes are the same that is.

Reply:
Given what we export in routed.cloud, those values should be replace with what is specified. However, now that you mention it, I'm not certain we want them overlapping, but I'm also unclear if we have the octets spare to split up what radvd is already handing out into multiple networks.